src/symbolize/gimli.rs

```diff
@@ -329,7 +328,9 @@ impl Mapping {
     }
 }
 
-thread_local! {
+type Cache = Vec<(PathBuf, Mapping)>;
+
+fn with_cache(f: impl FnOnce(&mut Cache)) {
```

Comment:
I think we'll want this to be `unsafe` with a doc comment saying it needs to be externally synchronized